client/components/place/placesOnMap.factory.js

```diff
@@ -98,8 +99,14 @@ angular.module('mapModule')
 
   placesOnMap.showTracks = function(tracksArray) {
     var trackForAdding;
+    var color;
     for (var i = 0; i < tracksArray.length; i++) {
```

Comment:
* use forEach and filters

smth like
```javascript
var getColor = function(track) {
    var type = tracksType.filter(function( trackType ) {
        return trackType.type == track.type;
    });
    return type.color;
}
```

* you can add some 3p libs like lodash